psychopy/iohub/util/__init__.py

```diff
@@ -255,45 +263,40 @@ def _getDevicePaths(iohub_device_path):
     scs_yaml_paths = []  # stores the paths to the device config files
     plugins.refreshBundlePaths()  # make sure eyetracker external plugins are reachable
 
-    # find EyeTracker classes by entry points targeting psychopy.iohub.devices.eyetracker
-    for group, points in importlib.metadata.entry_points().items():
-        # skip irrelevant groups
-        if not group.startswith("psychopy.iohub.devices.eyetracker"):
-            continue
-        # process each point
-        for ep in points:
-            # load class the entry point points to
+    # NOTE: The “selectable” entry points were introduced in importlib_metadata 3.6 and Python 3.10.
+    # Prior to those changes, entry_points accepted no parameters and always returned a dictionary
+    # of entry points, keyed by group. With importlib_metadata 5.0 and Python 3.12, entry_points
+    # always returns an EntryPoints object.
+
+    if 'eyetracker' in device_name.lower():
```

Comment:
This is because `device_name` gets passed down all the way from the ioHub util function [getDeviceNames()](https://github.com/psychopy/psychopy/blob/8a228c2f14c134ee81d74e92e0a8ad72e50eeded/psychopy/iohub/util/__init__.py#L359). Previously PsychoPy setting component just calls this function without argument, therefore forcing `device_name` to be `eyetracker.hw`. I've [updated this line](https://github.com/psychopy/psychopy/blob/8a228c2f14c134ee81d74e92e0a8ad72e50eeded/psychopy/experiment/components/settings/__init__.py#L46) to explicitly pass in the `device_name` argument to make it clear that this is used to load eyetrackers. This is why the mouse eyetracker implementation could be imported.

The reason for this if statement is because the `getDevicePaths()` function should be quite general, but we have this block of code that is hard-coded to look for entry-points related to eyetrackers. So I thought it's fair to check if `device_name` contains `eyetracker`. I checked everywhere where this ioHub function is used in PsychoPy, and this should produce a reasonable behavior.

My concern of not adding this `if 'eyetracker' in device_name.lower():` check is that if someone uses the `getDevicePaths()` function with a specific `device_name` in the future. For example, if EEG systems got incorporated into PsychoPy, then no matter what specific `device_name` I request, this function will also return all of the eyetracker config files, which feels like a buggy behavior to me. But let me know if you think otherwise and would like `getDevicePaths()` without argument to return ALL device configs. In that case, you can just change this line into
```
if 'eyetracker' in device_name.lower() or not device_name:
```